features/src/main/java/uk/gov/onelogin/features/login/ui/signin/welcome/WelcomeScreenViewModel.kt

```diff
@@ -79,12 +86,32 @@ class WelcomeScreenViewModel @Inject constructor(
                 },
                 onFailure = {
                     Log.e(tag, it?.message, it)
-                    navigator.navigate(LoginRoutes.SignInError, true)
+                    handleLoginErrors(it, fragmentActivity)
                 }
             )
         }
     }
 
+    private fun CoroutineScope.handleLoginErrors(
+        it: Throwable?,
+        fragmentActivity: FragmentActivity
+    ) {
+        this.launch {
+            when (it) {
+                is AuthenticationError -> {
+                    when (it.type) {
+                        AuthenticationError.ErrorType.ACCESS_DENIED -> {
```

Comment:
Can we lift this when check up one, so it's not nested?
something like:
```suggestion
                        when (it) {
                             is AuthenticationError.ErrorType.ACCESS_DENIED -> {
                             ....
                             }
                             else -> navigator.navigate(LoginRoutes.SignInError, true)
```

Reply:
Not sure if I'm correct, but we are not only catching AuthenticationErrors here, we are catching the app integrity ones as well, which are not Authentication Errors - if we lift it, we need to car it to a Authentication error to get teh type to check if it's access_denied, which is why I did it as above - does that make sense?
I can lift the second nested when into it's own function, if it makes it easier to read/ makes more sense?

Reply:
tech-debt ticket raised to remove complexity https://govukverify.atlassian.net/browse/DCMAW-12031